components/patient-access-rules/ui/src/main/resources/PhenoTips/PatientLock.xml

```diff
@@ -0,0 +1,397 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+-->
+
+<xwikidoc version="1.1">
+  <web>PhenoTips</web>
+  <name>PatientLock</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <creationDate>1433535415000</creationDate>
+  <parent>PhenoTips.PatientClass</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <date>1433886792000</date>
+  <contentUpdateDate>1433886501000</contentUpdateDate>
+  <version>1.1</version>
+  <title>Patient Lock</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>false</hidden>
+  <content>{{velocity}}
+#set ($patientId = $!{request.getParameter('patientId')})##The patient id to be locked or unlocked
+#set ($lockingRequest = $!{request.getParameter('lock')})##The locking request. True to lock, false to unlock.
+#set ($resultSuccessful = false)
+#if ("$!{patientId}" != '' &amp;&amp; "$!{lockingRequest}" != '')
+  #set ($resultSuccessful = $services.recordLocking.setRecordLock($patientId, $lockingRequest))
+#end
+#if ($resultSuccessful)
+  $response.setStatus(200)
+#else
+  $response.setStatus(500)
```

Comment:
`500` is an internal server error, specifically an error in the code. This branch is not usually an unknown server error, it's a bad request from the client. Maybe the return type should be changed from a simple boolean to an enum to indicate what exactly is wrong: unauthorized request, missing/wrong parameters, actual internal error.